src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/endpoints/JurisdictionEndpoint.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.endpoints;
+
+import feign.FeignException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Selector;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.Credential;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.JurisdictionToUserMapping;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.NoUserConfiguredException;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Map.Entry;
+
+@Component
+@Endpoint(id = "jurisdictions")
+@EnableConfigurationProperties(JurisdictionToUserMapping.class)
+public class JurisdictionEndpoint {
+
+    private static final Logger log = LoggerFactory.getLogger(JurisdictionEndpoint.class);
+
+    private final Map<String, Credential> jurisdictions;
+    private final Function<String, Credential> credentialProvider;
+    private final IdamClient idamClient;
+
+    public JurisdictionEndpoint(
+        JurisdictionToUserMapping mapping,
+        IdamClient idamClient
+    ) {
+        jurisdictions = mapping.getUsers();
+        credentialProvider = mapping::getUser;
+        this.idamClient = idamClient;
+    }
+
+    @ReadOperation
+    public Map<String, HttpStatus> jurisdictions() {
+        return jurisdictions
+            .entrySet()
+            .stream()
+            .collect(Collectors.toMap(
+                Entry::getKey,
+                entry -> checkCredentials(entry.getKey(), entry.getValue())
+            ));
+    }
+
+    @ReadOperation
+    public HttpStatus jurisdiction(@Selector String jurisdiction) {
+        try {
+            return checkCredentials(jurisdiction, credentialProvider.apply(jurisdiction));
+        } catch (NoUserConfiguredException exception) {
+            return HttpStatus.NOT_FOUND;
+        }
+    }
+
+    private HttpStatus checkCredentials(String jurisdiction, Credential credential) {
+        try {
+            idamClient.authenticateUser(credential.getUsername(), credential.getPassword());
+
+            log.debug("Successful authentication");
+
+            return HttpStatus.OK;
+        } catch (FeignException exception) {
+            logException(jurisdiction, credential.getUsername(), exception);
```

Comment:
This catch doesn't represent an error situation for this endpoint (even though for the service as a whole it is). I think it would be best to log it as nothing more severe than warning and let the caller decide what to do with errors coming from IdAM.